java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -358,14 +358,29 @@ public void delete(final String rawPath) throws IOException {
      * @throws IOException If an IO exception has occurred.
      * @throws MantaClientHttpResponseException If a HTTP status code other than {@code 200 | 202 | 204} is encountered
      */
-    public void delete(final String rawPath,
-                       final MantaHttpHeaders requestHeaders) throws IOException {
+    public void delete(final String rawPath, final MantaHttpHeaders requestHeaders) throws IOException {
         Validate.notBlank(rawPath, "rawPath must not be blank");
-
         String path = formatPath(rawPath);
         LOG.debug("DELETE {}", path);
-
         httpHelper.httpDelete(path, requestHeaders);
+        final Integer pruneEmptyParentDepth = config.getPruneEmptyParentDepth();
+        if (pruneEmptyParentDepth != null && 0 != pruneEmptyParentDepth) {
+            PruneEmpytParentDirectoryStrategy.pruneParentDirectories(this, requestHeaders, path, pruneEmptyParentDepth);
+        }
+    }
+
+    /**
+     * Deletes several directories in manta.
+     * @param paths - A list of paths to be deleted.
+     * @param requestHeaders - headers can be null.
+     * @throws IOException If a HTTP status code other than {@code 200 | 202 | 204} is encountered
+     */
+    public void delete(final List<String> paths, final MantaHttpHeaders requestHeaders) throws IOException {
```

Comment:
I wouldn't recommend increasing the API surface for a convenience method like this. The potential failure modes for this method mean that users can't be sure which delete in the list failed.

I'd venture to guess you're trying to avoid the infinite recursion of `MantaClient#delete(String, MantaHttpHeaders)` and `PruneEmpytParentDirectoryStrategy#pruneParentDirectories(String)` calling each other indefinitely. If that's the case I'd recommend adding a package-private `MantaClient#delete(String, MantaHttpHeaders, Integer)` method which can perform a regular delete when the `Integer` argument is null and delegates to the strategy class otherwise. Then the strategy class can explicitly pass `null` to get the plain single-delete functionality. In that case the body of `MantaClient#delete(String, MantaHttpHeaders)` would just be `this.delete(path, headers, config.getPruneEmptyParentDepth())`